src/video/tinspire/SDL_tinspirevideo.c

```diff
@@ -288,6 +288,12 @@ static int NSP_SetColors(_THIS, int firstcolor, int ncolors, SDL_Color *colors)
 */
 static void NSP_VideoQuit(_THIS)
 {
+	if ( this->hidden->buffer ) {

```

Comment:
What I meant is that you can do just this, unconditionally:

```
SDL_free(this->hidden->buffer);
this->hidden->buffer = NULL;
SDL_free(this->hidden->buffer2);
this->hidden->buffer2 = NULL;
```

Reply:
Oh i see... Strange that it's not done this way when cleaning the buffer upon allocating it but i'll do just that.